### PR TITLE
[Merged by Bors] - feat(NumberTheory/Fermat): fermat_eq_fermat_sub_one_pow_two_add_one

### PR DESCRIPTION
**Pull request by mo271**
- [x] depends on: #17618
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
could also be called `fermat_add_one` I suppose

**Comment by acmepjz** [comment]
IMHO `fermat_succ` is better

**Comment by mo271** [comment]
In general I agree, but there are other formulas describing `fermat n + 1` in terms of `fermat n`

**Comment by acmepjz** [comment]
> In general I agree, but there are other formulas describing `fermat n + 1` in terms of `fermat n`

Then you have to `fermat_succ_eq_blah_blah`, or `fermat_succ_eq'` (primed naming scheme is strongly discouraged if you have several formulas for `fermat (n + 1)` I think).

File: Mathlib/NumberTheory/Fermat.lean [hunk 1]
```diff
@@ -54,6 +54,11 @@ theorem fermat_eq_prod_add_two (n : ℕ) : fermat n = (∏ k in range n, fermat
   rw [fermat_product, Nat.sub_add_cancel]
   exact le_of_lt <| two_lt_fermat _
 
+theorem fermat_eq_fermat_sub_one_pow_two_add_one (n : ℕ) :
```
Comment (mo271) [inline Mathlib/NumberTheory/Fermat.lean:57]:
```suggestion
theorem fermat_succ_eq_fermat_sub_one_pow_two_add_one (n : ℕ) :
```

**Comment by Ruben-VandeVelde** [comment]
Please hold for #17618

**Comment by mo271** [comment]
> Please hold for #17618

Rebased, should be good to go now.

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**